executor/runcexecutor/executor.go

```diff
@@ -173,6 +178,10 @@ func (w *runcExecutor) Exec(ctx context.Context, meta executor.Meta, root cache.
 	if meta.ReadonlyRootFS {
 		opts = append(opts, containerdoci.WithRootFSReadonly())
 	}
+
+	if w.cgroupParent != "" {
+		ctx = namespaces.WithNamespace(ctx, w.cgroupParent)
```

Comment:
Just set the `containerdoci.WithCgroup(filepath.Join(parent, id))` and avoid all this magic with weird side effects.